network.go

```diff
@@ -0,0 +1,365 @@
+//
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package virtcontainers
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/vishvananda/netlink"
+)
+
+// NetworkInterface defines a network interface.
+type NetworkInterface struct {
+	Name   string
+	IPAddr string
+}
+
+// NetworkInterfacePair defines a pair between TAP and virtual network interfaces.
+type NetworkInterfacePair struct {
+	Name             string
+	VirtNetInterface NetworkInterface
+	TAPNetInterface  NetworkInterface
+}
+
+// NetworkConfig is the network configuration related to a network.
+type NetworkConfig struct {
+	PluginName string
+
+	NetworkNS     string
+	NetworkNSPath string
+
+	NetworkPairs []NetworkInterfacePair
+}
+
+// NetworkType describes the type of network specification.
+type NetworkType string
+
+const (
+	// NoopNetworkType is the No-Op network.
+	NoopNetworkType NetworkType = "noop"
+
+	// CNINetworkType is the CNI network.
+	CNINetworkType NetworkType = "cni"
+)
+
+// Set sets a network type based on the input string.
+func (networkType *NetworkType) Set(value string) error {
+	switch value {
+	case "noop":
+		*networkType = NoopNetworkType
+		return nil
+	case "cni":
+		*networkType = CNINetworkType
+		return nil
+	default:
+		return fmt.Errorf("Unknown network type %s", value)
+	}
+}
+
+// String converts a network type to a string.
+func (networkType *NetworkType) String() string {
+	switch *networkType {
+	case NoopNetworkType:
+		return string(NoopNetworkType)
+	case CNINetworkType:
+		return string(CNINetworkType)
+	default:
+		return ""
+	}
+}
+
+// newNetwork returns a network from a network type.
+func newNetwork(networkType NetworkType) network {
+	switch networkType {
+	case NoopNetworkType:
+		return &noopNetwork{}
+	case CNINetworkType:
+		return &cni{}
+	default:
+		return &noopNetwork{}
+	}
+}
+
+// genRandomInterface generates locally a unique interface name
+func genRandomInterface(prefix string) (string, error) {
+	var random *rand.Rand
+	for i := 0; i < 100; i++ {
+		iface := fmt.Sprintf("%s_%x", prefix, random.Uint32())
```

Comment:
We don't need this function anymore, we will rely completely on libsnnet package instead.